lib/util/git/index.ts

```diff
@@ -893,3 +896,30 @@ export function getUrl({
     pathname: repository + '.git',
   });
 }
+
+export async function pushCommitAsRef(
+  commitSha: string,
+  refName: string
+): Promise<void> {
+  await git.raw(['update-ref', refName, commitSha]);
+  await git.raw(['push', '--force', 'origin', refName]);
+}
+
+const treeItemRegex = regEx(
+  /^(?<mode>\d{6})\s+(?<type>blob|tree)\s+(?<sha>[0-9a-f]{40})\s+(?<path>.*)$/
+);
+
+export async function listCommitTree(commitSha: string): Promise<TreeItem[]> {
+  const treeSha = (await git.raw(['cat-file', '-p', commitSha])).slice(5, 45);
+  const contents = await git.raw(['cat-file', '-p', treeSha]);
```

Comment:
Document this and any assumptions it relies on

Reply:
Can maybe be more sustainable to split the contents, filter for `tree x` and then extract `x`.